### DISCUSSION TITLE
Loosing wifi connection

### DISCUSSION THREAD

**Discussion by eigenphase** [sle118/squeezelite-esp32#226] [Q&A]
My MuseLuxe with squeezelite-esp32 works well ... until it looses Wifi connection. This happens for example when there is a power outage or I power cycle the wifi router. squeezelite does not seem connect any more. After I manually power cycle it, it re-connects to Wifi.

Since I am using the squeezelite as announcement speaker for home assistant this is a little bit a showstopper.

Can this be fixed somehow?

**Comment by 8OND007** [comment]
I've also a Muse Luxe with Squeezelite.
Wifi connection sucks. Very unstable. Also have to power off/on very frequent.
I wonder if we can replace/enhance the Wifi Antenne or add an external antenna. (drill hole in case and connect an external antenna).
Or does the ESP32 chip doesn't have an external antenna connection ?
The Muse Luxe was not a good investment.
Squeezelite seems to work best on it. (ESPHome wasn't a succes either). 
The Volume buttons and play/pause buttons don't work on it.
I don't like the fact that on every Wifi disconnect for a certain time, the ESP32 Squeezelite starts it's own AP, is it possible to disable this ?

**Reply by sle118** [reply]
This isn't the right place to discuss Muse Luxe issues. I am sure the manufacturer would be more than happy to help with recommendations on possible hardware modifications, etc. Some ESP32 modules do have an option for an external antenna, but I don't know which chip was used on the Luxe.

I happen to have one of these here so I erased it and flashed from the installer. I guarantee that the volume buttons work. Short press = Vol+ Vol- and long press= Next song, Prev song.

There's no way to disable the access point from automatically starting, and not adviseable as some devices are sometimes buried in enclosures, making it impossible to recover if the connection is lost. There seems to be an issue that prevents a device from properly reconnecting if its access point goes down for too long. I believe I have fixed it in one of the development branches so I'll see if I can retrofit to the current branch.

**Reply by eigenphase** [reply]
> There seems to be an issue that prevents a device from properly reconnecting if its access point goes down for too long. I believe I have fixed it in one of the development branches so I'll see if I can retrofit to the current branch.

That would be really amazing. I am plagued by this a lot because it makes my doorbell, chimes, alarm etc (that I built around the speaker) very unreliable.

I’ll also try  newer version. Do you have a pointer to the fix and/or which version would you recommend to install?

**Reply by sle118** [reply]
I am not sure where I left that before being pulled away by many personal things over the past few months. I will try my best to have a look sooner rather than later

**Reply by eigenphase** [reply]
Thank you, really really appreciate once you get the chance .

**Reply by sle118** [reply]
And then again, having a full time job and little time to guess things out, do you think you could do me a favor and capture some logs before and after the connection is lost? There are so many possible scenarios here that even with the best intention I might never really understand what happens in your specific case. Logs can only speed my analysis. Without them, there's little motivation to setup a complete test environment with a dedicated router and all. I have 6 players here and they are rock solid... So I need some input on what the heck is different at your place.

Thank you!
